Sources/Ink/Internal/CodeBlock.swift

```diff
@@ -13,16 +13,20 @@ internal struct CodeBlock: Fragment {
     private var code: String
 
     static func read(using reader: inout Reader) throws -> CodeBlock {
-        try require(reader.readCount(of: marker) == 3)
+        let startingMarkerCount = reader.readCount(of: marker)
+        try require(startingMarkerCount >= 3)
 
-        let language = reader.readUntilEndOfLine()
+        var language = reader.readUntilEndOfLine().drop(while: {char in char.isWhitespace})
```

Comment:
Is something like this feasible?

```swift
try reader.readCharacters(matching: \.isSameLineWhitespace, required: false)
let language = reader.read(until: \.isSamelineWhitespace, required: false)
try reader.readUntilEndOfLine()
```

It would mean adding the second `.read` function and adding the `required` parameter to the first (already done for others though)–but it should be much more straightforward and probably faster?

Reply:
This spec paragraph made me do a strict lead/trail trim for now.
> The line with the opening code fence may optionally contain some text
> following the code fence; this is trimmed of leading and trailing
> whitespace and called the [info string](@). If the [info string] comes
> after a backtick fence, it may not contain any backtick
> characters.  (The reason for this restriction is that otherwise
> some inline code would be incorrectly interpreted as the
> beginning of a fenced code block.)

Reply:
This is true, but the language name specifically is described in the spec as:

>An info string can be provided after the opening code fence. Although this spec doesn’t mandate any particular treatment of the info string, the first word is typically used to specify the language of the code block. In HTML output, the language is normally indicated by adding a class to the code element consisting of language- followed by the language name.

Which seems to be the more relevant bit for the purposes here. Most likely fine regardless and can be addressed later when more strict conformance is being tested on. Just thought I'd point this out.

Reply:
I like the idea of stripping this whitespace out pre-processing rather than post, if possible. `Reader` already has a function called `discardWhitespacesAndNewlines`, could we perhaps add another one called `discardWhitespaces()`, and call that before processing the `language`? That way we wouldn't have to go back and correct, which is inline with the `O(N)` goal of this project.

Reply:
I gave it a try and it is a bit cleaner.  I am still doing the tail trim separately for now.